frontend/src/component/insights/componentsChart/ProjectHealthChart/ProjectHealthChart.tsx

```diff
@@ -73,12 +76,17 @@ export const ProjectHealthChart: VFC<IProjectHealthChartProps> = ({
                     data: weeks.map((item) => ({
                         health: item.total
                             ? (
-                                  ((item.total - item.stale) / item.total) *
+                                  ((item.total -
```

Comment:
don't we get this value from the backend? Also is it how we calculate it in the backend?

Reply:
if we decide to keep it here I'd extract the equation into a small fn so that we don't have those deeply nested calculations that line up vertically due to spacing constraints in our linter/formatter

Reply:
Good questions! I don't know about the first one, but I can find out. And yeah, I'm happy to extract it. Was considering that for testing anyway, so

Reply:
You're right. We **do** get this value from the backend. Except it's not a required property on the object, so it can be undefined. However, I think that might be an oversight. It appears to build fine locally with the marking and I've submitted a [PR to see if it'll work against all the tests](https://github.com/ivarconr/unleash-enterprise/pull/1457) etc.

It does appear to be the same calculation that we do in the back end, though I haven't been able to find where we calculate it on the project-level when we insert the data (do you know?).

But that said, I don't think we can use the health value here because it doesn't give us the correct result. The items here are projects, right? So imagine this:

You have two projects, A and B. 
- A has 100 flags, of which all are stale. That gives it a health rating of 0.
- B has 1 flag, which is not stale. Its health rating is 100. 

Now, if you were to record each project's health rating, sum it up, and divide that by the number of projects ((0 +100) / 2), you'd get a 50% health rating. That doesn't sound like the correct health for the instance to me. That's the average of the projects

So if you instead tally up the total flags and run the calc we do today, you'd get (101-100) / 101 * 100, which is ~1%. That sounds more correct to me.

The problem is that if we just add the health from each project, we lose information on how many flags are in each project. I remember Andreas talking about this and adjusting the calculation based on that. If we were to use each individual item's health here, we'd have to weight them based on the number of flags in each project.

---

And if you're thinking "but Thomas, if you calculate health as (sum health) / (total flags), then it looks right in your example". And yeah, 100/101 ~= 1% (in fact, it's 0.99, which is also what the other calculation is).

Let's add a project C, then. 100 flags, 25 stale. Health = 75%.

Using health values we get (0 + 100 + 75) / 201 ~= 0.87.

Using the current approach, we get (201 - 100 - 25) / 201 * 100 ~= 37.8

So ... I don't think that works. I'm not about to write a proof of why it doesn't, but I think you get the gist. (Though if you think I'm wrong, please **do** tell me, because I'm definitely invested now 😄 )